### PR TITLE
Isar benchmark

### PR DESCRIPTION
**Pull request by vaind**
Based on and replaces #1

### CODE REVIEW

File: lib/isar_sync_executor.dart [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+          'readMany' + (benchmarkQualifier ?? ''), () => _box.getAllSync(ids)));
+
+  Future<void> removeMany(List<int> ids) => Future.value(
+        tracker.track('removeMany',
+            () => _store.writeTxnSync((Isar isar) => _box.deleteAllSync(ids))),
+      );
+
+  Future<List<T>> queryStringEquals(String val) async => Future.value(indexed
+      ? tracker.track('queryStringEquals', () {
+          // indexed queries must be case sensitive (?)
```
Comment (greenrobot-team) [inline lib/isar_sync_executor.dart]: Reading https://github.com/isar/isar#queries I understand that querying an indexed String is always case insensitive, no? Or differently: case sensitive queries are not supported when using `where()` conditions?

File: lib/model.dart [hunk 2]
```diff
@@ -48,6 +50,16 @@ class TestEntityPlain implements TestEntity {
 
   TestEntityPlain(this.id, this.tString, this.tInt, this.tLong, this.tDouble);
 
+  // TODO remove later; isar v0.4.0 needs a "public zero-arg constructor"
+  // NOTE: you also need to fix isar.g.dart after each generation to use this
```
Comment (greenrobot-team) [inline lib/model.dart:54]: Should this be left as is? Would it be possible to change the other databases to use the zero arg constructor?

File: lib/objectbox-model.json [hunk 3]
```diff
@@ -111,7 +111,7 @@
     },
     {
       "id": "5:6999419589241537986",
-      "lastPropertyId": "4:2940873808771910881",
+      "lastPropertyId": "5:1520307525834954271",
```
Comment (greenrobot-team) [inline lib/objectbox-model.json]: Any reason why the ObjectBox model file has changed? Should this be reverted?

**Review by greenrobot-team — COMMENTED** [review]
Potential issue with queries, leftover code. Anyhow LGTM (at least the stuff that is supported/works).

File: lib/isar_sync_executor.dart (hunk 1 above)
Comment (vaind) [reply lib/isar_sync_executor.dart]: Hmm, seems like that's the case... I'll update the code comments. Another point that would need to be resolved before comparing benchmarks.

File: lib/model.dart (hunk 2 above)
Comment (vaind) [reply lib/model.dart:54]: that would have performance impacts (though minor) - it's better to construct the object completely with properties

File: lib/objectbox-model.json (hunk 3 above)
Comment (vaind) [reply lib/objectbox-model.json]: some in-between changes. Doesn't hurt but I'll roll back.

**Comment by vaind** [comment]
@peterlazar1993 and @raphire08 you were previously interested in the Isar comparison - would you like to review this? It's probably getting merged on Monday.

**Comment by raphire08** [comment]
@vaind seems good to me

**Pull request merged**